snapcraft/tests/repo/test_base.py

```diff
@@ -0,0 +1,270 @@
+
+
+class FixXmlToolsTestCase(RepoBaseTestCase):
+
+    scenarios = [
+        ('xml2-config only should fix', {
+            'files': [
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xml2-config'),
+                    'content': 'prefix=/usr/foo',
+                    'expected': 'prefix=root/usr/foo',
+                },
+            ]
+        }),
+        ('xml2-config only should not fix', {
+            'files': [
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xml2-config'),
+                    'content': 'prefix=/foo',
+                    'expected': 'prefix=/foo',
+                },
+            ]
+        }),
+        ('xslt-config only should fix', {
+            'files': [
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xslt-config'),
+                    'content': 'prefix=/usr/foo',
+                    'expected': 'prefix=root/usr/foo',
+                },
+            ]
+        }),
+        ('xslt-config only should not fix', {
+            'files': [
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xslt-config'),
+                    'content': 'prefix=/foo',
+                    'expected': 'prefix=/foo',
+                },
+            ]
+        }),
+        ('xml2-config and xslt-config', {
+            'files': [
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xml2-config'),
+                    'content': 'prefix=/usr/foo',
+                    'expected': 'prefix=root/usr/foo',
+                },
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xslt-config'),
+                    'content': 'prefix=/usr/foo',
+                    'expected': 'prefix=root/usr/foo',
+                },
+            ]
+        }),
+        ('xml2-config and xslt-config should not fix', {
+            'files': [
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xml2-config'),
+                    'content': 'prefix=/foo',
+                    'expected': 'prefix=/foo',
+                },
+                {
+                    'path': os.path.join('root', 'usr', 'bin', 'xslt-config'),
+                    'content': 'prefix=/foo',
+                    'expected': 'prefix=/foo',
+                },
+            ]
+        }),
+    ]
+
+    def test_fix_xml_tools(self):
+        for test_file in self.files:
+            path = test_file['path']
+            os.makedirs(os.path.dirname(path), exist_ok=True)
+            with open(path, 'w') as f:
+                f.write(test_file['content'])
+
+        BaseRepo('root').normalize()
+
+        for test_file in self.files:
+            self.assertThat(
+                test_file['path'], FileContains(test_file['expected']))
+
+
+class FixShebangTestCase(RepoBaseTestCase):
+
+    scenarios = [
+        ('python bin dir', {
```

Comment:
python2 and python3... I noticed this while refactoring and was going to fix in a separate PR